src/Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv/Internal/Listener.cs

```diff
@@ -46,51 +46,99 @@ private UvStreamHandle CreateListenSocket()
             switch (EndPointInformation.Type)
             {
                 case ListenType.IPEndPoint:
-                case ListenType.FileHandle:
-                    var socket = new UvTcpHandle(Log);
-
-                    try
-                    {
-                        socket.Init(Thread.Loop, Thread.QueueCloseHandle);
-                        socket.NoDelay(EndPointInformation.NoDelay);
-
-                        if (EndPointInformation.Type == ListenType.IPEndPoint)
-                        {
-                            socket.Bind(EndPointInformation.IPEndPoint);
-
-                            // If requested port was "0", replace with assigned dynamic port.
-                            EndPointInformation.IPEndPoint = socket.GetSockIPEndPoint();
-                        }
-                        else
-                        {
-                            socket.Open((IntPtr)EndPointInformation.FileHandle);
-                        }
-                    }
-                    catch
-                    {
-                        socket.Dispose();
-                        throw;
-                    }
-
-                    return socket;
+                    return ListenTcp(false);
```

Comment:
nit: use named arguments, `useFileHandle: ...`, here and for other calls to ListenTcp and ListenPipe.